### DISCUSSION TITLE
[QUESTION] Theming

### DISCUSSION THREAD

**Discussion by ghost** [benbusby/whoogle-search#275] [Q&A]
Hi,

Thanks for creating this. Works great. Sorry, I have many questions about theming on docker:

Can the webpage's title be changed? How?

Can the webpage's favicon be changed? How?
Unfortunately mounting and changing /usr/src/app/app/static/img/logo.png doesn't change the logo, what is the correct path?
(Is there a way to just map all the static img to a mount volume?)

How to add margin/padding to each search result? (Changing .ZINbbc xpd O9g5cc uUPGi {} doens't work)
Is there a way to hide the text footer but not the "next page button"?
How to change the purple border of the suggested search result

How to change the font size and line height in the result page? (Changing them under body {} didn't work even with !important. What should be the correct css?)

Not related to theming but is there a way to go to the direct link of images under image search? 
Not related to theming but I'm getting two related search somehow. One is in the middle of the search result page; one is at the bottom.

Thank you!

**Comment by benbusby** [comment]
> Can the webpage's title be changed? How?

> Can the webpage's favicon be changed? How?

>  Is there a way to hide the text footer but not the "next page button"?

No, not currently.

> Unfortunately mounting and changing /usr/src/app/app/static/img/logo.png doesn't change the logo, what is the correct path?
(Is there a way to just map all the static img to a mount volume?)

The logo is an svg loaded in by Flask with the `logo.html` template, so replacing it would be a bit more involved than just replacing the image. You could update that template to just have an `img` element of your own I guess.

> How to change the font size and line height in the result page? 

Looks like `.vvjwJb` is (currently) responsible for result title font, and `.s3v9rd` for result body font.

> How to add margin/padding to each search result?

`.ZINbbc` is (currently) the class for search result divs. Adding margin/padding to that should work. 

> Not related to theming but is there a way to go to the direct link of images under image search?

There's a PR open for that, but I haven't had time to review it yet.

> Not related to theming but I'm getting two related search somehow. One is in the middle of the search result page; one is at the bottom.

As in two of your search results are identical? Whoogle doesn't actually do anything to generate search results, those all come directly from Google, so that's some other issue. Let me know if I'm misunderstanding.

In the meantime I'm going to convert this to a discussion and hopefully others can chime in if there's anything else that I'm missing.

**Comment by ghost** [comment]
Thanks for your reply!

One more: How to change the purple border of the suggested search result? Thank you!

**Reply by benbusby** [reply]
Not currently possible, but being discussed in another issue for future resolution. 

Edit:

Well, not possible using the available variables. You should be able to override it directly though. The existing css is

```css
.autocomplete-items {
    border: 1px solid #685e79;
}
```

so specifying something else should change it.

**Reply by ghost** [reply]
Thanks! .autocomplete-items works with !important.
